olive/evaluator/metric.py

```diff
@@ -138,5 +76,67 @@ def validate_goal(cls, v, values):
         return v
 
 
+class Metric(ConfigBase):
+    name: str
```

Comment:
will the identifier for each type-subtype metric be `{name of Metric}-{name of subtype}`?

Reply:
maybe we can move the unique identifier `name` into `SubTypeMetric`? The SubTypeMetric with have fields `name` which is a unique id and `type`/`sub_type`.

Reply:
We will keep the hierarchical structure for Metric definition. But flatten the results in Signal results.